### PR TITLE
preserve parentheses for JSDoc type casting with extra spaces a…

### LINKED ISSUE #6121: JSDoc typedef parentheses are removed if a typedef ends with `**/` instead of `*/`
**Prettier 1.17.1**

**Before:**
```
foo = /** @type {!Foo} **/(bar);
```

**After:**
```
foo = /** @type {!Foo} **/ bar;
```

Playground link [here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAIhArgIwBs4MAqDASyjNNQB0p0MB1AJ2gHMGAzCCDALwZUpcgAEYATwAOJYAEIAYnwC+NVAAp8AQ1YBKANwMmAYQitWcMDB59Bw0Rgky5S1WU079RqCAA0IBDSMBTQAM7IoLrsAO4ACroIESjaAG4QFAAm-iD4rNpgANZwMADK0gVUHMgwrLhwAQAWMAC2hMyNFPBhFWBwpUldFKldksjgYREBVGFwrDBx+Rwt2sjc2oSzAQBWYQAeAEL5RSWl2i1wADJUcGsbWyC7e6VVxACKuBDwd5sNIBWsWascY6fCSQjQHLSVhUGDMLIwRrIAAcAAYAtCILM2NppONoXAgalbgEAI6feCLILJEDaMIAWigcDgmRZOUs5IolkW2mWqyQ61+AVmLQoNTqfzCrzgHy+twF9z+MG0+HhmURyAATAFatoKIQqmYWitxlBoCSQLhZgAVFXJQWzFQqIA)

Related to #4799

### PR DESCRIPTION
**Pull request by evilebottnawi**
…erisk at end

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

fixes #6121

### CODE REVIEW

File: tests/comments_jsdoc_typescript/type.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const foo1 = /** @type {!Foo} */(bar);
```
Comment (thorn0) [inline tests/comments_jsdoc_typescript/type.js]: Please rename `comments_jsdoc_typescript/type.js` → `comments_closure_typecast/extra_star.js`

Comment (evilebottnawi) [reply tests/comments_jsdoc_typescript/type.js]: Here extra space too, maybe `extra_spaces_and_asterisks.js`

Comment (thorn0) [reply tests/comments_jsdoc_typescript/type.js]: ok

Comment (evilebottnawi) [reply tests/comments_jsdoc_typescript/type.js]: I think better name is `type-casting-with-extra-spaces-and-asterisks.js` (long, but reflects everything)

Comment (thorn0) [reply tests/comments_jsdoc_typescript/type.js]: The folder is already named `comments_closure_typecast`.

Comment (evilebottnawi) [reply tests/comments_jsdoc_typescript/type.js]: oh, yes, my mistake

Comment (evilebottnawi) [reply tests/comments_jsdoc_typescript/type.js]: fixed

File: changelog_unreleased/javascript/pr-xxxx.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#### Preserve parentheses for JSDoc type casting with extra spaces and asterisk at end ([#7011](https://github.com/prettier/prettier/pull/7011) by [@evilebottnawi](https://github.com/evilebottnawi))
```
Comment (thorn0) [inline changelog_unreleased/javascript/pr-xxxx.md]: The file should be named pr-7011.md.

Comment (evilebottnawi) [reply changelog_unreleased/javascript/pr-xxxx.md]: :+1:

**Pull request merged**